source/MaterialXCore/Definition.h

```diff
@@ -100,6 +100,24 @@ class NodeDef : public InterfaceElement
         return getAttribute(NODE_ATTRIBUTE);
     }
 
+    /// Return the element's output type.
+    const string& getType() const override
+    {
+        size_t numActiveOutputs = getActiveOutputs().size();
+        if (numActiveOutputs > 1)
```

Comment:
Guess this won't be correct if you allow multiple outputs where the parent and child have the same output.

Reply:
Why is that? With the set solution that Niklas proposed and I'm working on it should be fine, no?

Reply:
Yes. Wrote this before the proposal.